app/ui/legacy/src/test/java/com/fsck/k9/ui/helper/RelativeDateTimeFormatterTest.kt

```diff
@@ -0,0 +1,108 @@
+package com.fsck.k9.ui.helper
+
+import android.os.SystemClock
+import com.fsck.k9.Clock
+import com.fsck.k9.RobolectricTest
+import com.nhaarman.mockitokotlin2.whenever
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.util.TimeZone
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito
+import org.robolectric.RuntimeEnvironment
+import org.robolectric.annotation.Config
+
+@Config(qualifiers = "en")
+class RelativeDateTimeFormatterTest : RobolectricTest() {
+
+    private val context = RuntimeEnvironment.application.applicationContext
+
+    private val clock = Mockito.mock(Clock::class.java)
+    private val zoneId = "Europe/Berlin"
+    private val now = LocalDateTime.parse("2020-05-17T15:42")
+    private val nowInMillis = now.toEpochMilli()
+
+    private val dateTimeFormatter = RelativeDateTimeFormatter(context, clock)
+
+    @Before
+    fun setUp() {
+        TimeZone.setDefault(TimeZone.getTimeZone(zoneId))
+        SystemClock.setCurrentTimeMillis(nowInMillis) // Is handled by ShadowSystemClock
+        whenever(clock.time).thenReturn(nowInMillis)
+    }
+
+    @Test
+    fun formatMessageDate_asTime_now() {
+        val displayDate = dateTimeFormatter.formatMessageDate(nowInMillis)
+
+        assertEquals("3:42 PM", displayDate)
+    }
+
+    @Test
+    fun formatMessageDate_asTime_earlier_today() {
+        val messageDate = now.minusHours(6).toEpochMilli()
```

Comment:
The custom parsing format was the only thing I didn't cover. Not because it would've been too hard to implement, but because the ISO date format is well-known by programmers and I think this is the target group here. But if you think a custom parser would be better, I can work on this as well.

Reply:
> But if you think a custom parser would be better, I can work on this as well.

It's fine, we don't have to overengineer that one :slightly_smiling_face: I agree that developers should have no trouble understanding the ISO date format. The idea behind my suggestion was that white space helps humans to more quickly parse the date/time.